federation/ldap/src/main/java/org/keycloak/storage/ldap/mappers/UserAttributeLDAPStorageMapper.java

```diff
@@ -453,14 +471,23 @@ private boolean isReadOnly() {
 
     protected void setPropertyOnUserModel(Property<Object> userModelProperty, UserModel user, String ldapAttrValue) {
         if (ldapAttrValue == null) {
+            if (Boolean.class.equals(userModelProperty.getJavaClass()) || boolean.class.equals(userModelProperty.getJavaClass())) {
+                userModelProperty.setValue(user, false);
+            } else {
+                userModelProperty.setValue(user, null);
+            }	
             userModelProperty.setValue(user, null);
         } else {
             Class<Object> clazz = userModelProperty.getJavaClass();
 
             if (String.class.equals(clazz)) {
                 userModelProperty.setValue(user, ldapAttrValue);
             } else if (Boolean.class.equals(clazz) || boolean.class.equals(clazz)) {
-                Boolean boolVal = Boolean.valueOf(ldapAttrValue);
+   		Boolean boolVal = false;
```

Comment:
Should be better to keep "Boolean boolVal = Boolean.valueOf(ldapAttrValue)" . That should handle the case when ldapAttrValue is "TRUE" or "true" (probably better regarding compatibility among various LDAP servers)

Reply:
Agree. Updated it.